entity-framework/core/modeling/value-conversions.md

```diff
@@ -2,145 +2,746 @@
+
+A value converter can be used to trim the padding when reading key values. This can be combined with the value comparer in the previous example to compare fixed length case-insensitive ASCII keys correctly. For example:
+
+<!--
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                var converter = new ValueConverter<string, string>(
+                    v => v,
+                    v => v.Trim());
+                
+                var comparer = new ValueComparer<string>(
+                    (l, r) => string.Equals(l, r, StringComparison.OrdinalIgnoreCase),
+                    v => v.ToUpper().GetHashCode(),
+                    v => v);
+
+                modelBuilder.Entity<Blog>()
+                    .Property(e => e.Id)
+                    .HasColumnType("char(20)")
+                    .HasConversion(converter, comparer);
+
+                modelBuilder.Entity<Post>(
+                    b =>
+                        {
+                            b.Property(e => e.Id).HasColumnType("char(20)").HasConversion(converter, comparer);
+                            b.Property(e => e.BlogId).HasColumnType("char(20)").HasConversion(converter, comparer);
+                        });
+            }
+-->
+[!code-csharp[ConfigureFixedLengthStrings](../../../samples/core/Modeling/ValueConversions/FixedLengthStrings.cs?name=ConfigureFixedLengthStrings)]
+
+### Encrypt property values
+
+Value converters can be used to encrypt property values before sending them to the database, and then decrypt them on the way out. For example, using string reversal as a substitute for a real encryption algorithm:
+
+<!--
+                modelBuilder.Entity<User>().Property(e => e.Password).HasConversion(
+                    v => new string(v.Reverse().ToArray()),
+                    v => new string(v.Reverse().ToArray()));
+-->
+[!code-csharp[ConfigureEncryptPropertyValues](../../../samples/core/Modeling/ValueConversions/EncryptPropertyValues.cs?name=ConfigureEncryptPropertyValues)]
+
+> [!NOTE]
+> There is currently no way to get a reference to the current DbContext, or other session state, from within a value converter. This limits the kinds of encryption that can be used. Vote for [GitHub issue #11597](https://github.com/dotnet/efcore/issues/12205) to have this limitation removed.
+
+> [!WARNING]
+> Make sure to understand all the implications if you roll your own encryption to protect sensitive data. Consider instead using pre-built encryption mechanisms, such as [Always Encrypted](/sql/relational-databases/security/encryption/always-encrypted-database-engine) on SQL Server.
+
+### Column facets and mapping hints
```

Comment:
This seems less like an example, and more like it should be part of the main documentation somewhere above.

Reply:
Agreed. WiIll move.